### ISSUE TITLE
Improvement for Alert KubeletTooManyPods? 

### ISSUE THREAD

**Issue by ulikl** [kubernetes-monitoring/kubernetes-mixin#223]
Currently a fix value is used for the maximum number of pods per node:
```
alert: 'KubeletTooManyPods',
            expr: |||
              kubelet_running_pod_count{%(kubeletSelector)s} > %(kubeletPodLimit)s * 0.9
||| % $._config,
```
But Kubernetes allows to set it per node.

This alert rule could be made a bit more cluster specific via :

```
max (kubelet_running_pod_count{job="kubelet"}) by (instance)   < on () group_left() max(kube_node_status_capacity_pods{job="kube-state-metrics"}) * 0.9
``` 
where kube_node_status_capacity_pods is the number of allowed pods per node from job kube-state-metrics.
Note: the value of kube_node_status_capacity_pods is typically lower for master and infra nodes.

This rule unfortunately cannot be defined on the exact limit per node as the label "node" from the "kube-state-metrics" is not available for metrics from other jobs.
I only found that the node IP and node name can be matched by using relabel_configs from kubernetes_sd_configs with role "node" instead of "endpoint" for kubelet.
Then the node label can be added from the node label "__meta_kubernetes_node_name". The node IP can be extracted from label "address" or label "instance".

Kind Regards,
Ulrike

**Comment by brancz** [comment]
Yes there is already a PR out there for this https://github.com/kubernetes-monitoring/kubernetes-mixin/pull/135. We should pick that up as 1.14 has been out for a while with the metric we needed for this.

**Comment by ulikl** [comment]
Ok. I haven't seen this.

I'm using OpenShift 3.11. Will you still consider an improvement for "release 0.1"?

**Comment by brancz** [comment]
OpenShift 3.11 won't ever be able to have this unfortunately as it doesn't have the metric required to make this work (as mentioned in the PR it will only work with Kubernetes 1.14 and up).

**Comment by simonpasquier** [comment]
I feel this has been resolved in the current version of kubernetes-mixin:

https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/e24e6c4322eafc22511364771ff8b71fda3b3df2/alerts/kubelet.libsonnet#L38-L50

**Comment by ulikl** [comment]
Yes, should fix it for Kubernetes 1.14 and higher.
Closing is ok for me.
